Remove mentions of defunct build platforms and runtimes

jonjensen opened the pull request:
BeOS, MS-DOS, and OS/2 haven't been supported by Ruby for a while.

IronRuby and MagLev are listed under "Discontinued" at https://github.com/planetruby/awesome-rubies

IronRuby last release was March 2011. Repo archived. (Only IronPython still seems to be alive.)

MagLev last release was October 2011.

Cardinal last commit was August 2013 (and last Parrot release in Feb. 2016).

riseshia commented on ko/about/index.md:
```diff
@@ -134,9 +134,9 @@ end
 
 * 루비는 운영체제에 의존하지 않는 스레딩 기능을 제공합니다. 그리고 이는 루비가 동작하는 모든 플랫폼에 적용되어서 스레드를
-  지원하지 않는 MS-DOS에서도 다중 스레드 프로그램을 작성할 수 있습니다.
+  지원하지 않는 에서도 다중 스레드 프로그램을 작성할 수 있습니다.
```
```suggestion
  지원하지 않을 때에도 다중 스레드 프로그램을 작성할 수 있습니다.
```

jonjensen replied:
Thank you for the correction!

riseshia commented on ko/about/index.md:
```diff
@@ -134,9 +134,9 @@ end
+  지원하지 않는 에서도 다중 스레드 프로그램을 작성할 수 있습니다.
 
-* 루비는 이식성이 매우 높습니다. 대부분 GNU Linux에서 개발되었지만 대부분의 UNIX, macOS, Windows, DOS, BeOS, OS/2 등에서 동작합니다.
+* 루비는 이식성이 매우 높습니다. 대부분 GNU/Linux에서 개발되었지만 대부분의 UNIX, macOS, Windows, 등에서 동작합니다.
```
```suggestion
* 루비는 이식성이 매우 높습니다. 대부분 GNU/Linux에서 개발되었지만 대부분의 UNIX, macOS, Windows 등에서 동작합니다.
```

bugtender commented on zh_tw/about/index.md:
```diff
@@ -110,9 +110,9 @@ Ruby 還具有以下的特點：
+* Ruby 具有與作業系統無關的多線程（threading）能力。可以在所有可以執行 Ruby 的平台上都能夠達到多線程的目標，而不必管作業系統是否支援。
 
-* Ruby 具有高度的移植性：它大部份是在 GNU/Linux 上發展出來，但是可以執行於多種的作業系統如： UNIX、macOS、Windows、DOS、BeOS、OS/2 等。
+* Ruby 具有高度的移植性：它大部份是在 GNU/Linux 上發展出來，但是可以執行於多種的作業系統如： UNIX、macOS、Windows、等。
```
There's an unnecessary listing comma.

```suggestion
* Ruby 具有高度的移植性：它大部份是在 GNU/Linux 上發展出來，但是可以執行於多種的作業系統如： UNIX、macOS、Windows 等。
```

jcserracampos commented:
PT translations is fine!

jonjensen commented:
> LGTM with zh_tw and zh_cn.

@bugtender GitHub's approval list still says: "ruby/www-ruby-lang-org-i18n-zh_cn was requested for review" - do you know how that can be marked accepted?

hlcfan commented:
Looking good, I added a comment regarding removing the trailing `、` yesterday. You already removed that 👍

enderahmetyurt commented on tr/about/index.md:
```diff
@@ -166,12 +166,11 @@ bahsedilmiştir:
 * Ruby yüksek taşınabilirliğe sahiptir. GNU/Linux üzerinde geliştirilmiş
   olsa dahi, UNIX’in bir çok çeşidi, macOS,
-  Windows, DOS, BeOS, OS/2, vb. üzerinde çalışmaktadır.
+  Windows, vb. üzerinde çalışmaktadır.
```
@jonjensen `vb.` means is `etc` in Turkish. is there any operating systems except Unix distributions, macOSX and Windows? In English it is `It runs under Linux and other UNIX-like operating systems, macOS and Windows` there is no etc so we can use `vb.` here. It should be in Turkish `Ruby yüksek taşınabilirliğe sahiptir. GNU/Linux üzerinde geliştirilmiş olsa dahi, UNIX’in bir çok çeşidi, macOS ve Windows üzerinde çalışmaktadır.`

jonjensen replied:
Ok, makes sense.

nagachika commented on en/about/index.md:
```diff
@@ -170,10 +170,10 @@ Ruby has a wealth of other features, among which are the following:
 * Ruby features OS independent threading. Thus, for all platforms on
   which Ruby runs, you also have multithreading, regardless of if the OS
-  supports it or not, even on MS-DOS!
+  supports it or not.
```
Hi.
The current ruby (maybe from 1.9 and later) depend on pthread/WinThread to implement Thread.
The whole description in this section is totally false for recent ruby and could be removed.

jonjensen replied:
Ok, that's fine with me. It was already gone from the French version. I removed it from all the others.

yous commented on ko/about/index.md:
```diff
@@ -135,7 +135,7 @@ Ruby는 개발자의 편의를 위해 많은 기능을 제공합니다. 그 중
 * Ruby는 운영체제에 의존하지 않는 스레딩 기능을 제공합니다. 그리고 이는 Ruby가 동작하는 모든 플랫폼에 적용되어서 스레드를
   지원하지 않는 MS-DOS에서도 다중 스레드 프로그램을 작성할 수 있습니다.
```
The changes are reverted.
```suggestion
  지원하지 않을 때에도 다중 스레드 프로그램을 작성할 수 있습니다.
```

hsbt replied:
Thanks!

gfvcastro commented:
Hey @jonjensen and @ruby/www-ruby-lang-org-editorial, what's the status of this pull request? I can work on the Spanish translations of it.

gfvcastro commented:
@jonjensen I agree with you. @ruby/www-ruby-lang-org-editorial, what do you think?

zzak commented:
@gfvcastro @jonjensen if someone wants to rebase and include Nagachika-san's suggestion, I think it could be approved:
https://github.com/ruby/www.ruby-lang.org/pull/2123/files#r330827913

jonjensen commented:
> @gfvcastro @jonjensen if someone wants to rebase and include Nagachika-san's suggestion, I think it could be approved: https://github.com/ruby/www.ruby-lang.org/pull/2123/files#r330827913

Ok, rebased, and updated to remove outdated threading notes as per Nagachika-san.

shyouhei commented on ja/about/index.md:
```diff
@@ -174,13 +171,6 @@ MRI以外のRuby処理系には以下のようなものがあります。
 
-#### 以下のバージョンはここ数年ほとんどまたは全く更新されていませんが、引き続き利用可能です。
-
-* [Rubinius][rubinius]は「Rubyで書かれたRuby」です。LLVM上に構築されています。LLVMは他の言語も構築されるよくできた仮想マシンです。
```
I guess deletion of this line is not intentional, because no other translations in this pull request does the same thing.

jonjensen replied:
@shyouhei It was intentional. The other translations don't have that at all. They didn't separate frequently-updated from less-updated.

I removed that line because all the less-frequently-updated ones are totally dead now.

shyouhei replied:
OK, got it.  Then you might want also want to delete the line starts with `[rubinius]` below.  That would become an orphan link definiton.

jonjensen replied:
@shyouhei This is done.

zzak replied:
>  The other translations don't have that at all. They didn't separate frequently-updated from less-updated.

This isn't entirely true, see:

https://github.com/ruby/www.ruby-lang.org/blob/eb6223c62011b571b826bf84e6a72b217d36475e/en/about/index.md?plain=1#L189

and:

https://github.com/ruby/www.ruby-lang.org/blob/eb6223c62011b571b826bf84e6a72b217d36475e/ko/about/index.md?plain=1#L149

yous commented:
Can you apply changes of `documentation/faq/` to `ko/` as well? `ko/documentation/faq/` articles were translated quite recently.

jonjensen commented:
> Can you apply changes of `documentation/faq/` to `ko/` as well? `ko/documentation/faq/` articles were translated quite recently.

@yous I'm not sure who you are asking there, but no, I'm not the one to update the Korean FAQ version. Someone else will need to do that.

yous commented on es/about/index.md:
```diff
@@ -230,11 +220,7 @@ del 2003.
 [tiobe]: http://www.tiobe.com/index.php/content/paperinfo/tpci/index.html
 [jruby]: http://jruby.org
-[rubinius]: http://rubini.us
+[rubinius]:	https://github.com/rubinius/rubinius
```
I think this is a typo.

```suggestion
[rubinius]: https://github.com/rubinius/rubinius
```

zzak commented:
This PR is a lesson on why in-repo translations are not a good idea.

Let's just fix the english version and let other people open PRs to fix the other languages.
